pygsti/extras/errorgenpropagation/propagatableerrorgen.py

```diff
@@ -0,0 +1,363 @@
+                          
+    elif ErG1.getType()=='C' and ErG2.getType()=='H':
+        errorGens = commute_errors(ErG2,ErG1,weightFlip=-1.0,BCHweight=BCHweight)
+    
+    elif ErG1.getType()=='H' and ErG2.getType()=='A':
+        pVec1 = com(ErG1.getP1() , ErG2.getP1())
+        errorGens.append( propagatableerrorgen('A'  , [pVec1[1] , ErG2.getP2()] , -1j*wT*pVec1[0]) )
+        pVec2 = com(ErG1.getP1() , ErG2.getP2())
+        errorGens.append( propagatableerrorgen('A'  , [ErG2.getP1(), pVec2[1]] , -1j*wT*pVec2[0] ) )
+                          
+    elif ErG1.getType()=='A' and ErG2.getType()=='H':
+        errorGens = commute_errors(ErG2,ErG1,weightFlip=-1.0,BCHweight=BCHweight)
+
+    elif ErG1.getType()=='S' and ErG2.getType()=='S':
+        errorGens.append( propagatableerrorgen('H', ErG1.getP1(),0 ))
+                         
+    elif ErG1.getType()=='S' and ErG2.getType()=='C':
+       pVec1=labelMultiply(ErG1.getP1() , ErG2.getP1())
+       pVec2=labelMultiply(ErG2.getP2() , ErG1.getP1())
+       errorGens.append( propagatableerrorgen( 'A' , [pVec1[1], pVec2[1]] , -1j*wT*pVec1[0]*pVec2[0]  ))
+       pVec1 = labelMultiply(ErG1.getP1() , ErG2.getP2())
+       pVec2 = labelMultiply(ErG2.getP1() , ErG1.getP1())
+       errorGens.append( propagatableerrorgen( 'A', [pVec1[1], pVec2[1]] , -1j*wT*pVec1[0]*pVec2[0]))
+       pVec1 =acom(ErG2.getP1(), ErG2.getP2())
+       pVec2 = labelMultiply(pVec1[1],ErG1.getP1())
+       errorGens.append( propagatableerrorgen( 'A'  ,[pVec2[1], ErG1.getP1()] , -1j*.5*wT*pVec1[0]*pVec2[0]))
+       pVec1=acom(ErG2.getP1(), ErG2.getP2())
+       pVec2=labelMultiply(ErG1.getP1(),pVec1[1])
+       errorGens.append( propagatableerrorgen( 'A',   [ErG1.getP1() ,pVec2[1]],-1j*.5*wT*pVec1[0]*pVec2[0]))
+                         
+    elif ErG1.getType() == 'C' and ErG2.getType() == 'S':
+       errorGens = commute_errors(ErG2,ErG1,weightFlip=-1.0,BCHweight=BCHweight)
+                         
+    elif ErG1.getType() == 'S' and ErG2.getType() == 'A':
+       pVec1 =labelMultiply(ErG1.getP1() , ErG2.getP1())
+       pVec2=labelMultiply(ErG2.getP2() , ErG1.getP1())
+       errorGens.append( propagatableerrorgen( 'C', [pVec1[1], pVec2[1]] ,1j*wT*pVec1[0]*pVec2[0] ))
+       pVec1=labelMultiply(ErG1.getP1() , ErG2.getP2())
+       pVec2=labelMultiply(ErG2.getP1() , ErG1.getP1())
+       errorGens.append( propagatableerrorgen( 'C', [pVec1[1], pVec2[1]] , -1j*wT*pVec1[0]*pVec2[0]))
+       pVec1 = com(ErG2.getP1() , ErG2.getP2())
+       pVec2 = com(ErG1.getP1(),pVec1[1])
+       errorGens.append( propagatableerrorgen( 'A', [ErG1.getP1(), pVec2[1]] ,-.5*wT*pVec1[0]*pVec2[0]))
+                         
+    elif ErG1.getType() == 'A' and ErG1.getType() == 'S':
+       errorGens = commute_errors(ErG2,ErG1,weightFlip=-1.0,BCHweight=BCHweight)
+                         
+    elif ErG1.getType() == 'C' and ErG2.getType() == 'C':
```

Comment:
First, I'll just say that this is way more readable with the intermediate variables. It would be even more readable if `pVec*` were named things like `AP` or `acomAB` or things like that.

Note has

$$\large
\begin{align*}
[C_{A,B}, C_{P,Q}] &= -i\big(A_{AP,QB} + A_{AQ,BP} - A_{BP,QA} - A_{BQ,PA}\big)\\
        &-\frac{i}{2}\big(A_{[P,\{A,B\}],Q} + A_{[Q,\{A,B\}],P} + A_{[\{P,Q\},A],B} + A_{[\{P,Q\},B],A}\big)\\
        &+\frac{i}{4}H_{[\{A,B\},\{P,Q\}]}
\end{align*}
$$

but coded up is

$$\large
\begin{align*}
[C_{A,B}, C_{P,Q}] &= -i\big(A_{AP,QB} + A_{AQ,\textcolor{red}{PB}} \textcolor{red}{+} A_{BP,QA} \textcolor{red}{+} A_{BQ,PA}\big)\\
        &-\frac{i}{2}\big(A_{[P,\{A,B\}],Q} + A_{[Q,\{A,B\}],P} + A_{[\{P,Q\},A],B} + A_{[\{P,Q\},B],A}\big)\\
        &+\frac{i}{4}H_{[\{A,B\},\{P,Q\}]}
\end{align*}
$$

Which is correct? (Also we are recomputing anticommutators)